### PR TITLE
Fix crashes when a subcommand has no docstring

### LINKED ISSUE #44: Nubia crashes when a subcommand has no docstring
After applying the patch from [this gist:](https://gist.github.com/ylectric/87d90a0c4b5fa8d2f38742a228374166)
```
.venv ❯ env PYTHONPATH=(pwd) python example/nubia_example.py super-command do-stuff
Traceback (most recent call last):
  File "example/nubia_example.py", line 17, in <module>
    options=Options(persistent_history=False))
  File "/Users/afomin/github/python-nubia/nubia/internal/nubia.py", line 126, in __init__
    for cmd in self._plugin.get_commands():
  File "/Users/afomin/github/python-nubia/example/nubia_plugin.py", line 51, in get_commands
    AutoCommand(sample_commands.SuperCommand),
  File "/Users/afomin/github/python-nubia/nubia/internal/cmdbase.py", line 172, in __init__
    inspection.command.help
  File "/opt/homebrew/Cellar/python36/3.6.2+_254.20170915/Frameworks/Python.framework/Versions/3.6/lib/python3.6/textwrap.py", line 430, in dedent
    text = _whitespace_only_re.sub('', text)
TypeError: expected string or bytes-like object
```

### PR DESCRIPTION
**Pull request by toti1212**
If a subcommand no has a docstring, set an empty string by default.

This closes #44

### CODE REVIEW

**Comment by AhmedSoliman** [comment]
@toti1212 Do you think you can get this update to conform to what we discussed on #44?

**Comment by toti1212** [comment]
> @toti1212 Do you think you can get this update to conform to what we discussed on #44?

Sorry about the delay ✋I hope I have contributed with this, it shows the same error when the command doesn't have a docstring. Also, I tried adding some tests.

File: tests/inspection_test.py [hunk 1]
```diff
@@ -55,3 +55,15 @@ def my_function(self, arg1: str, argument_2: int):
...
+        @command
+        class SuperCommand:
+            """SuperHelp"""
+
+            @command
+            def my_function(self, arg1: str):
+                pass
+
+        data = inspect_object(SuperCommand)
+        self.assertEquals(0, len(data.subcommands))
```
Comment (ylectric) [inline tests/inspection_test.py]: Please add a newline at the end of the file

Comment (ylectric) [inline tests/inspection_test.py]: I think `self.assertListEqual([], data.subcommands)`

File: tests/supercommands_test.py [hunk 2]
```diff
@@ -80,3 +80,20 @@ def sub_command(self, arg1: str, arg2: int):
...
         )
+
+    def test_super_no_docstring(self):
+        @command
+        class SuperCommand:
+            "SuperHelp"
+
+            @command
+            def sub_command(self, arg1: str):
+                return "Hi {}".format(arg1)
```
Comment (ylectric) [inline tests/supercommands_test.py]: Please use f-strings instead.

File: nubia/internal/typing/__init__.py [hunk 3]
```diff
@@ -330,6 +332,16 @@ def inspect_object(obj, accept_bound_methods=False):
...
             metadata = inspect_object(candidate, accept_bound_methods=True)
+            # ignore subcommands without docstring
+            if not metadata.command.help:
+                cprint(
+                    (
+                        "[WARNING] The sub-command {} will not be loaded. "
+                        "Please provide a help message by either defining a "
+                        "docstring or filling the help argument in the "
+                        "@command annotation"
+                    ).format(metadata.command.name), "red")
```
Comment (ylectric) [inline nubia/internal/typing/__init__.py]: Please use [f-strings](https://www.python.org/dev/peps/pep-0498/) instead.

**Comment by toti1212** [comment]
🤦‍♂ sorry, I hadn't understood you. I will change that. ✋

**Comment by toti1212** [comment]
> @toti1212, by [f-strings](https://www.python.org/dev/peps/pep-0498/) I meant string interpolation, not `%`-style formatting. Here is an example:
> 
> ```
> Python 3.7.3 (default, Mar 27 2019, 09:23:15)
> Type 'copyright', 'credits' or 'license' for more information
> IPython 7.4.0 -- An enhanced Interactive Python. Type '?' for help.
> 
> In [1]: value = 42
> 
> In [2]: f"The value is {value}"
> Out[2]: 'The value is 42'
> ```
> 
> Could you please fix this?

🤦‍♂ sorry, I hadn't understood you. I will change that. ✋

**Comment by toti1212** [comment]
> Accepted, thanks!
> Before I land it: have you followed the instructions about `pre-commit`? In other words, do Black and/or flake8 have any complaints here?
> We probably should set them up as a part of Travis regardless of pre-commit hooks.

Proud to be able to contribute to this project! 🙌
Initially, I have not followed the instructions, but I had already configured my vs code with the same tools. In addition, I just configured pre-commit and nothing to change appeared 👍

**Comment by ylectric** [comment]
Hey @toti1212, this PR revealed some bugs in our internal codebase. I'll address them and then rebase & land your change.

**Comment by toti1212** [comment]
Hi @ylectric, write to me if it needs some changes.

**Comment by ylectric** [comment]
@toti1212, I've just submitted a fix for our internal codebase. Will rebase your PR as soon as it's approved.

**Pull request closed without merging**